packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1861,6 +1902,13 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             type: envelope.contents.type,
         };
 
+        // Only collect signal telemetry for messages sent by the current client.
+        if (message.clientId === this.clientId && envelope.clientSignalSequenceNumber !== undefined) {
```

Comment:
I do not thin envelope.clientSignalSequenceNumber !== undefined check is required - if it was sent by this client, then it has to have a sequence number. If you want to check it, I'd rather use assert - it tells readers better what are the intentions (and does validation).